From a01acbbebafffcbb44a458aa8e5b37d014d02080 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:08:46 -0400
Subject: [acpi] hotplug: Fix acpi_bus_get_device() return value check

Message-id: <20140710010846.22146.27079.stgit@gir.stowe>
Patchwork-id: 85639
O-Subject: [RHEL7.1 PATCH 100/105] ACPI / hotplug: Fix acpi_bus_get_device() return value check
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: 5beaee4f195d0297473da9a45459050b561e10cf

commit 5beaee4f195d0297473da9a45459050b561e10cf
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Thu Nov 14 00:54:00 2013 +0100

    ACPI / hotplug: Fix acpi_bus_get_device() return value check

    Since acpi_bus_get_device() returns a plain int and not acpi_status,
    ACPI_FAILURE() should not be used for checking its return value.  Fix
    that.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Acked-by: Toshi Kani <toshi.kani@hp.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/scan.c b/drivers/acpi/scan.c
index 856b092..be7a603 100644
--- a/drivers/acpi/scan.c
+++ b/drivers/acpi/scan.c
@@ -411,8 +411,7 @@ static void acpi_hotplug_notify_cb(acpi_handle handle, u32 type, void *data)
   break;
  case ACPI_NOTIFY_EJECT_REQUEST:
   acpi_handle_debug(handle, "ACPI_NOTIFY_EJECT_REQUEST event\n");
-  status = acpi_bus_get_device(handle, &adev);
-  if (ACPI_FAILURE(status))
+  if (acpi_bus_get_device(handle, &adev))
    goto err_out;
 
   get_device(&adev->dev);
-- 
1.7.1